src/EventSubscriber/RequestSubscriber.php

```diff
@@ -271,12 +283,63 @@ protected function constructQuerySpan($connection, $driver, array $query) {
    *   An array of options to pass to the PHP Agent. Ie. tags.
    */
   protected function prepareAgentOptions() {
+    $tags = [];
+
     $route_options = $this->routeMatch->getRouteObject()->getOptions();
     if (isset($route_options['_admin_route'])) {
-      return ['tags' => ['is_admin_route' => TRUE]];
+      $tags['is_admin_route'] = TRUE;
+    }
+
+    // Add tags based on the path patterns configured.
+    $tags = $tags + $this->preparePathPatternTags();
+
+    return $tags;
+  }
+
+  /**
+   * Provide tags based on the path patterns configured.
+   *
+   * @return array
+   *   An array of tags to pass to the PHP Agent.
+   */
+  protected function preparePathPatternTags() {
+    $tags = [];
+
+    // Fetch the current path from route object.
+    $route_object = $this->routeMatch->getRouteObject();
+    $path = $route_object->getPath();
+
+    // Fetch the configured path patterns.
+    $tag_config = $this->apiService->getTagConfig();
+    $path_pattern = $tag_config['path_patterns'];
+
+    if (empty($path_pattern)) {
+      return;
+    }
+
+    // Explode path patterns by new line.
+    $path_patterns = explode(PHP_EOL, $path_pattern);
+
+    // Add tags depending on the path pattern set.
+    foreach ($path_patterns as $path_pattern) {
+      $patterns = explode(':', $path_pattern);
```

Comment:
**CHANGE REQUEST**

Now this is where it all becomes confusing and bug-prone with variable names. What is the difference between `$path_pattern`, $path_patterns`, `$patterns`?

We need to do some refactoring here. Create a function that takes the configuration value and returns an array of already parsed pattern/tag pairs:

```
[
  [
    'pattern' => '/product/*',
    'tag_key' => 'provider',
    'tag_value' => 'commerce',
  ],
  [
    'pattern' => '/checkout/*',
    'tag_key' => 'provider',
    'tag_value' => 'commerce',
  ],
];
```

So:

```
$patterns = $this->parsePatterns($this->apiService->getTagConfig());
foreach ($patterns as $pattern) {
  if (!$this->pathMatcher->matchPath($path, $patterns['pattern'])) {
    continue;
  }

  $tags[$pattern['tag_key']] = $pattern['tag_value'];
}
```
The above will result in more readable and concise code without variable naming problems. You will also be able to reuse part of the code for processing route patterns as well.

Reply:
Added a new method `parsePatterns` in apiService class